src/core/segment_allocator.cc

```diff
@@ -12,10 +12,10 @@ constexpr size_t kSegmentShift = MI_SEGMENT_SHIFT;
 namespace dfly {
 
 SegmentAllocator::SegmentAllocator(mi_heap_t* heap) : heap_(heap) {
-  // mimalloc uses 4MiB segments and we might need change this code if it changes.
-  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3;
+  // mimalloc uses 32MiB segments and we might need change this code if it changes.
+  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3 + 3;
   static_assert(kSegLogSpan == kSegmentShift);
-  static_assert((~kSegmentAlignMask) == (MI_SEGMENT_SIZE - 1));
+  // static_assert((~kSegmentAlignMask) == (MI_SEGMENT_SIZE - 1));
```

Comment:
99% Sure this was a mistake. When this was introduced it would not pass the assertion on the previous versions. 

Also mimalloc uses 32MiB segments but we fetch 4MiB. We can bump this up to `8MiB` or `32MiB` to give 1-1 mapping.

I am not very familiar with `SmallString` which uses this extensively so I will update on this.